panel/pane/holoviews.py

```diff
@@ -354,6 +355,11 @@ def _cleanup(self, root: Model | None = None) -> None:
                 old_pane._cleanup(root)
         super()._cleanup(root)
 
+    def _server_destroy(self, session_context: BokehSessionContext) -> None:
+        with param.discard_events(self):
+            self.object = None
```

Comment:
This is not safe to do. While in general Panel components are created per session it is possible to reuse them across sessions so you cannot clean up the parameters in this way.

I'm also quite dubious about this actually causing a true memory leak. Certainly it can result in a circular reference being created that means that garbage collection has to do a full collection to free up the memory. If there is a true memory leak then this fix is simply reducing its severity.